test/net/sourceforge/kolmafia/request/ChateauRequestTest.java

```diff
@@ -0,0 +1,68 @@
+package net.sourceforge.kolmafia.request;
+
+import static internal.helpers.Networking.html;
+import static internal.helpers.Player.withProperty;
+import static internal.matchers.Preference.isSetTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import internal.helpers.Cleanups;
+import net.sourceforge.kolmafia.KoLCharacter;
+import net.sourceforge.kolmafia.preferences.Preferences;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+public class ChateauRequestTest {
+  @BeforeAll
+  public static void beforeAll() {
+    KoLCharacter.reset(true);
+    KoLCharacter.reset("ChateauRequestTest");
+    Preferences.reset("ChateauRequestTest");
+    Preferences.setBoolean("chateauAvailable", true);
```

Comment:
Do these using withProperty() Cleanups.

Reply:
Made an attempt, let me know if this is the pattern you were thinking of.

Reply:
Here's what I would do.

ChateauRequestTest is for testing everything about the Chateau.
You are testing free rests in particular.

The pattern I would like is this:

```
public class ChateauRequestTest {

  @BeforeAll
  public static void beforeAll() {
    KoLCharacter.reset("ChateuRequestTest");
    Preferencesr.reset("ChateuRequestTest");
  }

  @BeforeEach
  public void beforeEach() {
    // stuff
  }

  @Nested
  class FreeRest {

    public Cleanups propertyCleanups() {
      // ... stuff that applies to free rests
      return cleanups;
    }

    @Test
    public void test1() {
      var cleanups = new Cleanups(propertyCleanups());
      try (cleanups) {
        // test code
      }
    }
  }
}
```

Reply:
Restructured the test - I think this is the right way to do it. Let me know if you disagree.

Reply:
(And thanks for the help - it's been at least a decade since I used JUnit...)

Reply:
Very nice!